agent.go

```diff
@@ -404,6 +427,19 @@ func NewAgent(config *AgentConfig) (*Agent, error) {
 	return a, nil
 }
 
+func (a *Agent) getCandidateTypesForConfig(config *AgentConfig) ([]CandidateType, error) {
```

Comment:
We can remove. I would make users be explicit, I just want to keep this as simple as possible.